filebeat/module/elasticsearch/audit/ingest/pipeline.json

```diff
@@ -10,56 +10,29 @@
-                    "ES_AUDIT_EVENT_TYPE": "\\[%{WORD:event.type}\\]",
-                    "ES_AUDIT_ORIGIN_TYPE": "(origin_type\\=\\[%{WORD:elasticsearch.audit.origin_type}\\])?",
-                    "ES_AUDIT_ORIGIN_ADDRESS": "(origin_address\\=\\[%{IPORHOST:source.ip}\\])?",
-                    "ES_AUDIT_PRINCIPAL": "(principal\\=\\[%{WORD:user.name}\\])?",
-                    "ES_AUDIT_REALM": "(realm\\=\\[%{WORD:elasticsearch.audit.realm}\\])?",
-                    "ES_AUDIT_ROLES": "(roles\\=\\[%{DATA:elasticsearch.audit.roles}\\])?",
-                    "ES_AUDIT_ACTION": "(action\\=\\[%{DATA:elasticsearch.audit.action}(\\[%{DATA:elasticsearch.audit.sub_action}\\])?\\])?",
-                    "ES_AUDIT_URI": "(uri=\\[%{DATA:url.original}\\])?",
-                    "ES_AUDIT_INDICES": "(indices\\=\\[%{DATA:elasticsearch.audit.indices}\\])?",
-                    "ES_AUDIT_REQUEST": "(request\\=\\[%{WORD:elasticsearch.audit.request}\\])?",
-                    "ES_AUDIT_REQUEST_BODY": "(request_body\\=\\[%{DATA:http.request.body.content}\\])?"
-                },
                 "patterns": [
-                    "%{ES_TIMESTAMP}\\s*%{ES_NODE_NAME}\\s*%{ES_AUDIT_LAYER}\\s*%{ES_AUDIT_EVENT_TYPE}\\s*%{ES_AUDIT_ORIGIN_TYPE},?\\s*%{ES_AUDIT_ORIGIN_ADDRESS},?\\s*%{ES_AUDIT_PRINCIPAL},?\\s*%{ES_AUDIT_REALM},?\\s*%{ES_AUDIT_ROLES},?\\s*%{ES_AUDIT_ACTION},?\\s*%{ES_AUDIT_INDICES},?\\s*%{ES_AUDIT_URI},?\\s*%{ES_AUDIT_REQUEST},?\\s*%{ES_AUDIT_REQUEST_BODY},?"
-                ]
-            }
-        },
-        {
-            "split": {
-                "field": "elasticsearch.audit.roles",
-                "separator": ",",
-                "ignore_missing": true
-            }
-        },
-        {
-            "split": {
-                "field": "elasticsearch.audit.indices",
-                "separator": ",",
-                "ignore_missing": true
+                    "^%{CHAR:first_char}"
+                ],
+                "pattern_definitions": {
+                    "CHAR": "."
+                }
             }
         },
         {
-            "script": {
-                "lang": "painless",
-                "source": "if (ctx.elasticsearch.audit.sub_action != null) { ctx.elasticsearch.audit.action += '[' + ctx.elasticsearch.audit.sub_action + ']' }"
+            "pipeline": {
+                "if": "ctx.first_char != '{'",
+                "name": "{< IngestPipeline "pipeline-plaintext" >}"
             }
         },
         {
-            "remove": {
-                "field": "elasticsearch.audit.sub_action",
-                "ignore_missing": true
+            "pipeline": {
+                "if": "ctx.first_char == '{'",
+                "name": "{< IngestPipeline "pipeline-json" >}"
             }
         },
         {
             "date": {
-                "field": "elasticsearch.audit.timestamp",
+                "field": "elasticsearch.audit.@timestamp",
```

Comment:
Nit: prior to grabbing the real timestamp from the log, could you populate `event.created` with Beat's `@timestamp`?

Reply:
This is already being done in the very first processor in this pipeline. It's collapsed in the diff since nothing changed there:

https://github.com/elastic/beats/pull/10352/files#diff-a007340015953dcf31f9d48f74358ac3R4

Reply:
Missed that 👍